cardano-cli/app/cardano-cli.hs

```diff
@@ -40,4 +55,25 @@ main = toplevelExceptionHandler $ do
 
   co <- Opt.customExecParser dynamicPrefs (opts envCli)
 
-  orDie (docToText . renderClientCommandError) $ runClientCommand co
+  runRIO co (runClientCommand co) `catch` 
+    (\(e :: CustomCliException) -> do 
+      let commandText = case co of 
+            AnyEraCommand anyEraCommand  -> renderAnyEraCommand anyEraCommand 
+            AddressCommand addrCmds  -> renderAddressCmds addrCmds 
+            ByronCommand{}  -> "" 
+            CompatibleCommands anyCompatCmd -> renderAnyCompatibleCommand  anyCompatCmd
+            HashCmds hashCmds  -> renderHashCmds hashCmds 
+            KeyCommands keyCmds  -> renderKeyCmds keyCmds 
+            NodeCommands nodeCmds  -> renderNodeCmds nodeCmds 
+            QueryCommands queryCmds  -> renderQueryCmds queryCmds 
+            LegacyCmds legacyCmds  -> renderLegacyCommand legacyCmds 
+            CipFormatCmds cipFmtCmds  -> renderCipFormatCmds cipFmtCmds 
+            CliPingCommand{}  -> ""
+            CliDebugCmds{} -> "" 
+            Help{} -> "" 
+            DisplayVersion -> "" 
```

Comment:
I wonder whether it would be possible to deduce this generically from the `opts`, or something like that, it is odd that we have to pattern match for each case.

And well, obviously there are missing empty strings here `""`, which may be on purpose, but I am just telling you in case it is an overlook.

Reply:
We could write `Pretty` instances for the commands for that purpose.

Reply:
> I wonder whether it would be possible to deduce this generically from the opts, or something like that
I've thought this before but never looked closely into it.